PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1220,7 +1221,12 @@ ApplyFwImage (
     Status = UpdateSystemFirmware(ImageHdr);
     *ResetRequired = TRUE;
   } else if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gCsmeFWUpdateImageFileGuid) == TRUE) {
-    Status = UpdateCsme(CapImage, CapImageSize, ImageHdr);
+    if (FeaturePcdGet(PcdCsmeUpdateEnabled) == 1) {
+      CsmeUpdateInData = InitCsmeUpdInputData();
+      if (CsmeUpdateInData != NULL) {
+        Status = UpdateCsme(CapImage, CapImageSize, CsmeUpdateInData, ImageHdr);
+      }
+    }
     *ResetRequired = TRUE;
```

Comment:
If PcdCsmeUpdateEnabled is disabled, reset is required?

Reply:
corrected it